### ISSUE TITLE
Create Email Notifications for Manifold Activities

### ISSUE THREAD

**Issue by mkgold** [ManifoldScholar/manifold#951]
## Feature Description
A configurable email notification system (with optional daily/weekly digest modes) that will notify Manifold admins and editors about a) new projects; b) new texts uploaded to an instance; c) comments; d) annotations

## *Why* is this feature important? *Who* does it help?
This feature will help admins keep abreast of overall activity on a Manifold instance. It will also be useful to editors and authors who want to know about (and, perhaps, respond to) new comments and annotations

## User Stories
A Manifold instance admin grants an editor or project creator role to someone, but wants to keep track of what texts are uploaded to be sure that nothing is uploaded outside of copyright restrictions. An email notification of a new project creation or new text upload will remind the admin to look over the newly uploaded text.

An author whose text is uploaded wants to be notified of new comments on their text so that they can respond to reader queries, notes, and responses. The activity drives further commenting and interaction with the text.

## Design Notes
the settings should be configurable so that people can choose to receive notifications as the associated activity occurs or through a digest mode (daily or weekly). The settings should also be configurable on a per-project basis (ie, an editor may want email updates on one project but not another)

**Comment by tsmyre** [comment]
@mkgold Should there also be an email system for readers to opt into if they want notice when a project they are following is updated?

**Comment by mkgold** [comment]
ooh -- yes @tsmyre -- that is a good idea.

**Comment by SMaxOwok** [comment]
Also included in this will be the addition of a `COMMENT_CREATED` event and the refactor of event type constants to `ClassyEnum` class.

```
class EventType < ClassyEnum::Base
  # @abstract
  # @return [String]
  def digest_title
    raise NotImplementedError, "must implement #{self.class}::#{__method__}"
  end
end

class EventType::ProjectCreated < EventType
  def digest_title
    "Created #{owner.project.name}"
  end
end

# ...
```

**Comment by SMaxOwok** [comment]
Needs design for:

- [ ] Notification management form
- [ ] Successfully unsubscribed
- [ ] Notification emails

**Comment by SMaxOwok** [comment]
I think I'm getting close to a point where it will be useful to know what kind of information to include in the digest emails.  Thoughts?

**Comment by zdavis** [comment]
Hey @SMaxOwok can you please remind me where we're at with this? I know you did work on this, but I don't see a pull request.

**Comment by SMaxOwok** [comment]
It's mostly built out, but I was waiting on feedback about specifically what should be in the emails.  I thought @tsmyre's comment above was kind of the jumping off point for people to give their opinions on, since I remember going back and forth a lot about who wanted what.

**Comment by zdavis** [comment]
Hey @SMaxOwok, I know a lot of work has been done on this, and I'd like to get it in 2.0. Possible? I don't think anyone else is going to add feedback at this point, so perhaps we can go with what @tsmyre proposed?

**Comment by SMaxOwok** [comment]
Oh man... I can try.  Let me get it rebased on master and see what it looks like.  I'm not sure how much cleaning up/styling needs to be done.

**Comment by zdavis** [comment]
If not, it’s not the end of the world. If I had to choose between global
backend changes and this, I’d choose global backend changes.

**Comment by laelito** [comment]
Hi all -- so we've got some updated email styling to accompany this issue. Check out the attached PDF comp.
[Emails.pdf](https://github.com/ManifoldScholar/manifold/files/2271987/Emails.pdf)

**Comment by tsmyre** [comment]
I like this. Very clean and to the point. What is intended with the lorem ipsum blocks at the bottom? On pages 1 and 2 there is only one block and then two blocks on pages 3 and 4.

**Comment by mkgold** [comment]
I agree. Nice work @laelito !

@tsmyre thanks so much for speccing out all of these emails! I really like them all. I'd probably leave out "oh snap!" but otherwise I think this is great.

**Comment by mkgold** [comment]
Not sure whether to start a new ticket or not, but I've been looking at these reports and I would really love to see whether we can include info on project creator to these emails -- somewhere in the attached screenshot I would love to see something like "by [Project creator name]" so that I know who added the text in question. On a big instance like CUNY's that's useful information -- if it's a trusted user, I barely need to check; if it's someone new to the platform, I may want to take a look 

![screen shot 2019-02-13 at 9 31 46 pm](https://user-images.githubusercontent.com/278503/52758608-f1ef2800-2fd6-11e9-9b7e-8f13da603b0d.png)
